### PR TITLE
Organizations Listing 2: Electric Boogaloo

### PR DESCRIPTION
**Pull request by S4-K1**
I really hope this is the last update I need to do to this but chances are it's not

### CODE REVIEW

File: src/modules/organizations.ts [hunk 1]
```diff
@@ -15,16 +15,21 @@ import type { FieldData } from "./fields.js";
 export interface IOrganization extends Entity, TreeNode, Fields {
     rezkit_id: string;
     name: string;
+
+    parent_id: string | null;
+
+    children: IOrganization[];
```
Comment (LeoAdamek) [inline src/modules/organizations.ts]: This shouldn't be here. Organization does not implement TreeNode.

File: src/modules/organizations.ts [hunk 2]
```diff
@@ -15,16 +15,21 @@ import type { FieldData } from "./fields.js";
 export interface IOrganization extends Entity, TreeNode, Fields {
```
Comment (LeoAdamek) [inline src/modules/organizations.ts]: `IOrganization` should not implement `TreeNode` or `Fields` as the underlying data models do not implement these behaviours.

File: src/modules/organizations.ts [hunk 3]
```diff
@@ -39,6 +44,9 @@ export class Organization implements IOrganization {
     name!: string;
     rezkit_id!: string;
     readonly updated_at!: Date;
+    readonly children!: IOrganization[];
+    fields!: FieldData;
+    parent_id!: string | null;
```
Comment (LeoAdamek) [inline src/modules/organizations.ts]: Remove this as `TreeNode` and `Fields` should not be implemented.

File: src/modules/user.ts [hunk 4]
```diff
@@ -60,11 +61,6 @@ export interface OrganizationSettings {
 
 export type UpdateOrganizationSettings = Partial<OrganizationSettings>;
 
-export interface IOrganization extends Entity, OrganizationSettings {
```
Comment (LeoAdamek) [inline src/modules/user.ts]: Why did you replace the IOrganization model when we already have one that works?

**Review by LeoAdamek — CHANGES_REQUESTED** [review]
Needs some changes, the `IOrganization` interface is not correct.

File: src/modules/user.ts (hunk 4 above)
Comment (S4-K1) [reply src/modules/user.ts]: I was trying to be smart and move organization stuff into the organizations file, will move back

File: src/modules/organizations.ts [hunk 5]
```diff
@@ -53,7 +47,7 @@ export class Api extends ApiGroup {
     async list(params?: OrganizationListQuery): Promise<Paginated<Organization>> {
         const response = (await this.axios.get<Paginated<IOrganization>>(`/organizations`, { params })).data
 
-        response.data = response.data.map(o => new Organization(o, this.axios))
```
Comment (LeoAdamek) [inline src/modules/organizations.ts:56]: I don't think this latest change is correct, you still need to map each of the response items to an `Organization` instance.

**Review by LeoAdamek — CHANGES_REQUESTED** [review]
Almost there, just one small tweak.

File: src/modules/organizations.ts (hunk 5 above)
Comment (S4-K1) [reply src/modules/organizations.ts:56]: `response.data = response.data.map(new Organization(this.axios))`

Comment (S4-K1) [reply src/modules/organizations.ts:56]: Wait no

**Comment by S4-K1** [comment]
This entire pr was a waste of time because I misunderstood how to retrieve from the frontend

**Pull request closed without merging**